File: gsb/gsb-web/src/main/webapp/gsb/crm/sys/js/sys-salesman-tree-part.js
/**
 * Created by win on 2018/1/22.
 */
System.Declare("com.gongsibao.crm.web");

com.gongsibao.crm.web.SysSalesmanTreePart =org.netsharp.panda.commerce.TreePart.Extends( {

    ctor: function () {
        this.base();
    }
});


//重新调用查询
$(function() {
    var supplierId = controllerGsbCrmSysDepartmentTree.queryString('supplierId');
    var params = {
        supplierId: supplierId

    };
    $('#treeGsbCrmSysDepartmentTree').tree("options").queryParams = params;
    $('#treeGsbCrmSysDepartmentTree').tree('reload');

});

File: gsb/gsb-crm/src/main/java/com/gongsibao/crm/web/SysSalesmanTreePart.java
package com.gongsibao.crm.web;

import java.util.ArrayList;
import java.util.List;

import org.netsharp.panda.commerce.TreePart;
import org.netsharp.util.StringManager;

import com.gongsibao.utils.SupplierSessionManager;

/**
 * Created by win on 2018/1/22.
 */
public class SysSalesmanTreePart extends TreePart {

    @Override
    public String getExtraFilter() {

        List<String> ss = new ArrayList<String>();
        String filter = getRequest("filter");
        if (!StringManager.isNullOrEmpty(filter)) {

            return null;
        }
        String id = this.getRequest("id");
        if (StringManager.isNullOrEmpty(id)) {

            ss.add("(parent_id=0 or parent_id is null)");
        }else{
            ss.add("parentId="+id);
        }

        // 过滤服务商ID
        String requestSupplierId = this.getRequest("supplierId");
        if (!StringManager.isNullOrEmpty(requestSupplierId)) {

            ss.add("supplierId=" + requestSupplierId);
        } else {

            Integer supplierId = SupplierSessionManager.getSupplierId();
            if (supplierId != null) {

                ss.add("supplierId=" + supplierId);
            }
        }
        return StringManager.join(" and ", ss);
    }

}
